### ISSUE TITLE
Automatically retry tests

### ISSUE THREAD

**Issue by WesCossick** [sparksuite/w3c-css-validator#95]
Since W3C is a bit finicky at times causing tests to occasionally fail, we should add a `jest.retryTimes(2)` line to a `jest-setup.ts` file. This will allow all tests to automatically retry.

We'd need to add a `jest-setup.ts` file to the root of the project, and then reference it by adding this line to the `jest.config.ts` file:

```ts
setupFilesAfterEnv: ['<rootDir>/jest-setup.ts'],
```

**Comment by WesCossick** [comment]
@corymharper There may be other issues at play in the `v1.2.0` release's pipeline: https://github.com/sparksuite/w3c-css-validator/actions/runs/1295031970. Retrying several times doesn't seem to work.

**Comment by corymharper** [comment]
> @corymharper There may be other issues at play in the `v1.2.0` release's pipeline: https://github.com/sparksuite/w3c-css-validator/actions/runs/1295031970. Retrying several times doesn't seem to work.

There's some strange stuff going on in that pipeline. For example this test:

```
  ● #validateURL() › Throws when the timeout is passed

    The request took longer than 10000ms

      35 | 		// Listen for timeout event and reject in callback
      36 | 		req.on('timeout', () => {
    > 37 | 			reject(new Error(`The request took longer than ${timeout}ms`));
         | 			       ^
      38 | 		});
      39 | 	});
      40 | };
```

The code for that test is:

```ts
		it('Throws when the timeout is passed', async () => {
			await expect(
				validateURL(
					'https://rawcdn.githack.com/sparksuite/w3c-css-validator/6cf7b194b4f0b246678ed5101a2b6f0fb2918361/public/css/valid.css',
					{ timeout: 1 }
				)
			).rejects.toThrow('The request took longer than 1ms');
		});
```

To begin with it's fairly doubtful the timeout would have been passed at 10 seconds, but the more strange thing here is that the timeout is set explicitly to `1` ms, and there is no way that it could have been `10000` in that test. Here's the code that passes in the timeout:

```ts
const cssValidationResponse = await retrieveValidation(url, options?.timeout ?? 10000);
```

So I'm not a 100% sure how that error could be generated as printed. The test that timed out is explainable to me, but not the other.

**Comment by WesCossick** [comment]
Yeah, not 100% sure what's going on. I'll let you keep doing some digging and research...

**Comment by corymharper** [comment]
> Yeah, not 100% sure what's going on. I'll let you keep doing some digging and research...

So, I reviewed the failures of all previous attempts and the current attempt. The failures were never consistent at all, as in various tests could fail, but mostly all for the same reason: "Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout"

So, I decided that the most logical first step is to increase the test timeout to at least be greater than the default timeout for the methods of our packages. I introduced that functionality here: https://github.com/sparksuite/w3c-css-validator/pull/96

As for the inconsistent error message that showed up in one pipeline but none of the others, I was never able to reproduce it. I think it was a one off inconsistency, possibly even an example of a bit flip error.
